pkg/objstore/azure/helpers.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+
+	blob "github.com/Azure/azure-storage-blob-go/2018-03-28/azblob"
+)
+
+var (
+	blobFormatString = `https://%s.blob.core.windows.net`
+)
+
+type blobReadCloser struct {
+	Reader io.Reader
+}
+
+func (r *blobReadCloser) Read(b []byte) (n int, err error) {
+	return r.Reader.Read(b)
+}
+
+func (r *blobReadCloser) Close() error {
+	return nil
+}
+
+// DirDelim is the delimiter used to model a directory structure in an object store bucket.
+const DirDelim = "/"
+
+func getContainerURL(ctx context.Context, accountName, accountKey, containerName string) blob.ContainerURL {
+	c := blob.NewSharedKeyCredential(accountName, accountKey)
+	p := blob.NewPipeline(c, blob.PipelineOptions{
+		Telemetry: blob.TelemetryOptions{Value: "Thanos"},
+	})
+	u, err := url.Parse(fmt.Sprintf(blobFormatString, accountName))
+	if err != nil {
+		return blob.ContainerURL{}
+	}
+	service := blob.NewServiceURL(*u, p)
+	container := service.NewContainerURL(containerName)
+	return container
+}
+
+func getContainer(ctx context.Context, accountName, accountKey, containerName string) (blob.ContainerURL, error) {
+	c := getContainerURL(ctx, accountName, accountKey, containerName)
+
+	_, err := c.GetProperties(ctx, blob.LeaseAccessConditions{})
+	return c, err
+}
+
+func createContainer(ctx context.Context, accountName, accountKey, containerName string) (blob.ContainerURL, error) {
+	c := getContainerURL(ctx, accountName, accountKey, containerName)
+
+	_, err := c.Create(
+		context.Background(),
+		blob.Metadata{},
+		blob.PublicAccessNone)
+	return c, err
+}
+
+func getBlobURL(ctx context.Context, accountName, accountKey, containerName, blobName string) blob.BlockBlobURL {
+	container := getContainerURL(ctx, accountName, accountKey, containerName)
```

Comment:
I would consider inlining this.. 

The rule is following:
If you create a variable. It means that you want to use it more than once. if you do it just for one usage it confuses, because I am always automatically looking.. where else?